joomla/joomla-cms
# [5.0] Smart Search: Use UTF8-aware functions when indexing

Pull Request for Issue #40543.

### Summary of Changes
The parsing and processing code so far used non-utf8-safe functions like substr(), strrpos(), etc. This resulted in broken parsing in some cases and thus the behavior of #40543.


### Testing Instructions
Please copy the text from #40543 into an article and replace the p-tags between the paragraphs with br-tags instead. Save the content. If you look at the entry in #__finder_links for that content item, you will see the content in the description column and some words have been pushed together, removing the space between them.


### Actual result BEFORE applying this Pull Request
No space between some words.


### Expected result AFTER applying this Pull Request
All words are seperated with one space.


### Link to documentations
Please select:
